File: leetcode/Python/find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        sol, v = -1, [0]*(len(nums)+1)
        for n in nums:
            v[n] += 1
            if v[n] == 2:
                sol = n
                break
        return sol

File: leetcode/C++/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int sol = -1;
        vector<int> v(nums.size()+1,0);
        for(int &n : nums) if(++v[n]>1) {sol=n; break;}
        return sol;
    }
};